File: 04-FEA-Truss/Matlab-Code/PrintSolution.m
function PrintSolution(UF,PP,UUR,PUR)

fout = fopen('UF_Vector.dat','w');
fprintf(fout, '%12.8f \n', UF);
fclose(fout); 

fout = fopen('PP_Vector.dat','w');
fprintf(fout, '%12.8f \n', PP);
fclose(fout); 

fout = fopen('UUR_Vector.dat','w');
fprintf(fout, '%12.8f \n', UUR);
fclose(fout); 

fout = fopen('PUR_Vector.dat','w');
fprintf(fout, '%12.8f \n', PUR);
fclose(fout); 

fprintf('NODE DISPLACEMENT\n')
disp(UUR);
fout = fopen('UUR_Vector.dat','w');
for i = 1:size(UUR,1)
    fprintf(fout, '%12.8f ', UUR(i,:));
    fprintf(fout, '\n');
end

fprintf('NODE FORCE\n')
disp(PUR);
fout = fopen('PUR_Vector.dat','w');
for i = 1:size(PUR,1)
    fprintf(fout, '%12.8f ', PUR(i,:));
    fprintf(fout, '\n');
end


end

File: 02-FEA-1D/C-Code/PrintSolution.cpp
#include <MainFile.h>
using namespace std;

void PrintSolution(arma::colvec& Uf, arma::colvec& Pp,  arma::colvec& Uur,   arma::colvec& Pur)
{

  ofstream fout;

  fout.open ("UF_Vector.dat");
  fout << Uf;
  fout.close();
 
  fout.open ("PP_Vector.dat");
  fout << Pp;
  fout.close();

  fout.open ("UUR_Vector.dat");
  fout << Uur;
  fout.close();

  fout.open ("PUR_Vector.dat");
  fout << Pur;
  fout.close();


  cout<<"NODE DISPLACEMENT\n";
  cout<<Uur<<endl;
  cout<<"NODE FORCE\n";
  cout<<Pur<<endl;

}
